bag/src/main/java/com/trempel/bag/ui/BagViewModel.kt

```diff
@@ -0,0 +1,97 @@
+package com.trempel.bag.ui
+
+import androidx.databinding.library.baseAdapters.BR
+import androidx.lifecycle.*
+import com.trempel.bag.R
+import com.trempel.bag.model.toBagDomainModel
+import com.trempel.bag.model.toBagEntity
+import com.trempel.core_network.bag_db.db.BagDbRepository
+import com.trempel.bag.repository.BagNetworkRepository
+import com.trempel.core_ui.RecyclerItem
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class BagViewModel @Inject constructor(
+    private val bagNetworkRepository: BagNetworkRepository,
+    private val bagDbRepository: BagDbRepository
+) : ViewModel(), DefaultLifecycleObserver {
+
+    private val _bagItems = MutableLiveData<List<RecyclerItem>>()
+    val bagItems: LiveData<List<RecyclerItem>> get() = _bagItems
+    val quantity = MediatorLiveData<Int>()
+
+    private val _total = MutableLiveData<Float>()
+    val total: LiveData<Float> get() = _total
+
+    private val deleteObserver = Observer<Int> { productId ->
+        viewModelScope.launch {
+            bagDbRepository.deleteProductFromDbBagById(productId)
+            _bagItems.value = _bagItems.value
+                ?.filter { (it.data as BagItemViewModel).item.id != productId }
+        }
+    }
+
+    init {
+        quantity.addSource(_bagItems) {
+            quantity.value = getQuantitySum()
+            _total.value = getFinalPrice()
+        }
+    }
+
+    override fun onStart(owner: LifecycleOwner) {
+        super.onStart(owner)
+        loadData()
+    }
+
+    private fun loadData() {
+        viewModelScope.launch {
+            _bagItems.value = bagDbRepository.getProductsFromDbBag()
+                .map { bagNetworkRepository.getProduct(it.productId) to it.quantity }
```

Comment:
I think we shouldn't merge/map data models on the presenter level, let's keep it on the data layer